infrastructure/assumptions/non-local-ports.sub.window.js

```diff
@@ -0,0 +1,63 @@
+// Verifies that non-local HTTP(S) ports are open and serve correctly.
+//
+// See the corresponding WPT RFC:
+// https://github.com/web-platform-tests/rfcs/blob/master/rfcs/address_space_overrides.md
+//
+// These ports are used to test the Private Network Access specification:
+// https://wicg.github.io/private-network-access/
+//
+// More tests can be found in `fetch/private-network-access/`.
+
+// Resolves a URL relative to the current location, returning an absolute URL.
+//
+// `url` specifies the relative URL, e.g. "foo.html" or "http://foo.example".
+// `options.protocol` and `options.port`, if defined, override the respective
+// properties of the returned URL object.
+function resolveUrl(url, options) {
+  const result = new URL(url, window.location);
+  if (options === undefined) {
+    return result;
+  }
+
+  const { port, protocol } = options;
+  if (port !== undefined) {
+    result.port = port;
+  }
+  if (protocol !== undefined) {
+    result.protocol = protocol;
+  }
+
+  return result;
+}
+
+promise_test(() => {
+  const url = resolveUrl("resources/exists.txt", {
```

Comment:
```suggestion
  const url = resolveUrl("/common/blank.html", {
```

Since the content isn't used, might as well use an existing empty file.

Reply:
Good idea, thanks!

Reply:
Actually I needed to add `Access-Control-Allow-Origin: *` to allow accessing `response.ok` on the cross-origin response, so I kept the `resources/exists.txt` file.